### PR TITLE
Close peerstore and document Host Close

### PR DESCRIPTION
**Pull request by aarshkshah1992**
For https://github.com/libp2p/go-libp2p/issues/1030.

### CODE REVIEW

**Comment by MichaelMure** [comment]
LGTM :+1:

**Comment by aarshkshah1992** [comment]
ping @Stebalien

File: libp2p.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func ChainOptions(opts ...Option) Option {
 // - If no peerstore is provided, the host is initialized with an empty
 // peerstore.
 //
-// Canceling the passed context will stop the returned libp2p node.
```
Comment (raulk) [inline libp2p.go:55]: Is this the only correct order to call these actions? What happens if we call Close before we cancel the context?

Comment (aarshkshah1992) [reply libp2p.go:55]: @raulk The order is NOT important. I've updated the PR.

**Comment by aarshkshah1992** [comment]
ping @Stebalien @raulk  for approval.

**Review by Stebalien — APPROVED** [review]
I seem to remember that we did it this way because we don't "own" the peerstore, it may be shared. However, closing this is probably strictly better than what we currently do.

Nit: I'd use:

```go
if pstore := h.Peerstore(); pstore != nil {
    pstore.Close()
}

**Pull request merged**